enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/SegmentFile.java

```diff
@@ -262,4 +255,15 @@ public long size()
     {
         return fileSystem.getFileSize( file );
     }
+
+    @Override
+    public void close() throws DisposedException
+    {
+        closeWriter();
+        if ( !markedForDisposal )
+        {
+            markForDisposal( () -> {
```

Comment:
This will set the disposal flag to true, and signals that this segment file should be deleted. This is not what you want to do.